joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/util/TypeInfoCalculator.scala

```diff
@@ -0,0 +1,185 @@
+package io.joern.javasrc2cpg.util
+
+import com.github.javaparser.ast.`type`.{PrimitiveType, Type}
+import com.github.javaparser.resolution.declarations.{
+  ResolvedDeclaration,
+  ResolvedEnumConstantDeclaration,
+  ResolvedMethodDeclaration,
+  ResolvedTypeDeclaration,
+  ResolvedTypeParameterDeclaration
+}
+import com.github.javaparser.resolution.types.{
+  ResolvedArrayType,
+  ResolvedPrimitiveType,
+  ResolvedReferenceType,
+  ResolvedType,
+  ResolvedTypeVariable,
+  ResolvedVoidType
+}
+import com.github.javaparser.symbolsolver.javaparsermodel.declarations.JavaParserClassDeclaration
+import com.github.javaparser.symbolsolver.model.typesystem.{LazyType, NullType}
+import com.github.javaparser.symbolsolver.reflectionmodel.{ReflectionClassDeclaration, ReflectionTypeParameter}
+import io.joern.javasrc2cpg.util.TypeInfoCalculator.{TypeConstants, TypeNameConstants}
+import io.joern.x2cpg.datastructures.Global
+
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+class TypeInfoCalculator(global: Global) {
+  private val resolveCache = mutable.HashMap.empty[Type, Option[ResolvedType]]
+
+  def name(typ: ResolvedType): String = {
+    nameOrFullName(typ, false)
+  }
+
+  def fullName(typ: ResolvedType): String = {
+    registerType(nameOrFullName(typ, true))
+  }
+
+  private def nameOrFullName(typ: ResolvedType, fullyQualified: Boolean): String = {
+    typ match {
```

Comment:
I'm not sure that this is where the tests failed, but a default case should be added here.